### DISCUSSION TITLE
How to have both serverless and edge work for prisma version 5

### DISCUSSION THREAD

**Discussion by quan612** [prisma/prisma#22111] [Q&A] [answered]
### Question

Anyone have a working example for prisma@5 and nextjs that in the same repo, both serverless and edge works?
Here is what we did but not sure if works:

- Having a prismaProvider.ts file which import from  "@prisma/client" (this points to the DIRECT_DATABASE_URL ~ posgresql://) 

- Having a prismaEdfeProvider.ts file which import from  "@prisma/client/edge" (this points to the DATABASE_URL ~ prisma://)

- Having an api route as serverless, import from prismaProvider.ts

- Having an api route as edge, import from prismaEdfeProvider.ts

Scenario A:

In package.json file,
`"vercel-build": "npx prisma migrate deploy && next build && prisma generate && prisma generate --no-engine"`

This will yield an error on running the site with this error:
```
PrismaClientKnownRequestError: 
Invalid `prisma.[tablename].findMany()` invocation:
Error validating datasource `db`: the URL must start with the protocol `prisma://`
```

Scenario B:
We decided to omit the --no-engine in package.json
`"vercel-build": "npx prisma migrate deploy && next build && prisma generate"`

 Even though we get the response back from the edge api route, there are a bunch of error like below in the logs
and the queries are still being counted as accelerated, to be billed in usage on [console.prisma.io](http://console.prisma.io/)

Environment:
Nextjs 13
Prisma@5.5.2 on posgresql 14 (edited)

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

![image](https://github.com/prisma/prisma/assets/14017471/cdbc4b2a-2a44-41b4-996a-f6d6c9147d91)

**Comment by ludralph** [comment, marked as answer]
Hi @quan612 👋 

Thank you for raising this question.

To use Prisma in Edge environment, you would need to make use of [Accelerate](https://www.prisma.io/data-platform/accelerate) which will generate a connection string for your database that begins with `prisma://` and your import would use `@prisma/client/edge`. For serverless functions, you should use `@prisma/client`. We are also working on making the import simpler by eliminating the need to import from `@prisma/client/edge`, In the future hopefully you will be able to use @prisma/client import for edge environments as well.
For now, your scenario B solution should work. The messages that you are observing in the console are warnings and not errors.
By skipping the `--no-engine flag`, the overhead you are having is that the bundle size is more in size as the engine files are included.

You should note that we are actively working to have Prisma Client support in Edge environment. This can be seen from this [issue](https://github.com/prisma/prisma/issues/18763)

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
